addons/sale_project/tests/test_sale_project.py

```diff
@@ -160,3 +160,30 @@ def test_sol_product_type_update(self):
 
         self.product_order_service3.type = 'service'
         self.assertTrue(sale_order_line.is_service, "As the product is a service, the SOL should be a service")
+
+    @users('demo')
+    def test_cancel_so_linked_to_project(self):
+        """ Test that cancelling a SO linked to a project will not raise an error """
+        # Ensure user don't have edit right access to the project
+        group_sale_manager = self.env.ref('sales_team.group_sale_manager')
+        group_project_user = self.env.ref('project.group_project_user')
+        self.env.user.write({'groups_id': [(6, 0, [group_sale_manager.id, group_project_user.id])]})
+
+        partner = self.env['res.partner'].create({'name': "Mur en brique"})
```

Comment:
It could be defined in setUpClass method to avoid create the same partner in 2 tests. 🙂